### PR TITLE
CONSOLE: visually distinguish fixed from float baselines

### LINKED ISSUE #101: Console: plot fixed baseline solutions in different color than float
or other means of distinguishing them.
@denniszollo how much work do you think this would be?

### PR DESCRIPTION
**Pull request by henryhallam**
Closes #101, sorta
@denniszollo @fnoble this look ok to you?

### CODE REVIEW

**Comment by denniszollo** [comment]
I'd rather do the dual color thing

**Comment by henryhallam** [comment]
2015-07-06-141832_1520x882_scrot

Crosshair is now only shown for fixed solutions, not float ones.

**Comment by cbeighley** [comment]
I'd vote for the crosshair being shown on both.

File: piksi_tools/console/baseline_view.py [hunk 1]
```diff
@@ -208,23 +215,36 @@ def baseline_callback(self, sbp_msg):
...
+    self.ds = self.ds[-self.plot_history_max:]
+    self.fixeds = self.fixeds[-self.plot_history_max:]
+
+    self.plot_data.set_data('n_fixed', np.array(self.ns)[np.where(self.fixeds)])
+    self.plot_data.set_data('e_fixed', np.array(self.es)[np.where(self.fixeds)])
+    self.plot_data.set_data('d_fixed', np.array(self.ds)[np.where(self.fixeds)])
+    floats = np.logical_not(self.fixeds)
+    self.plot_data.set_data('n_float', np.array(self.ns)[floats])
+    self.plot_data.set_data('e_float', np.array(self.es)[floats])
+    self.plot_data.set_data('d_float', np.array(self.ds)[floats])
```
Comment (fnoble) [inline piksi_tools/console/baseline_view.py]: I could see this having some performance impact, did you notice any slowdown when you are at `pol_history_max` points?

**Comment by fnoble** [comment]
Apart from the performance question this looks good, no worse than the existing code at least :)

File: piksi_tools/console/baseline_view.py (hunk 1 above)
Comment (henryhallam) [inline piksi_tools/console/baseline_view.py]: I didn't, but I was testing on a PC with plenty of CPU.  I'll run it with htop on the Atom tablet and see how it looks.

**Comment by henryhallam** [comment]
@cbeighley do you think the crosshair has to change color or are you OK with it being black for both fixed and float?

**Comment by fnoble** [comment]
@cbeighley : do you think it could be confusing to have the corsshair on both? I imagine someone looking at the float crosshair that isn't updating and thinking something was wrong

**Comment by fnoble** [comment]
c.f. RTKLIB, they only use one crosshair only it changes colour

**Comment by fnoble** [comment]
I we do have two though, I think they should be coloured the same colour as their traces (like the SPP tab)

**Comment by cbeighley** [comment]
Don't mind if it's two colors or one, but I think it's good to have an obvious indicator of where the last float solution was.

**Comment by henryhallam** [comment]
Review appreciated
Performance seems OK but could always use more testing
2015-07-09-150153_956x1076_scrot
2015-07-09-150137_956x1076_scrot

**Comment by cbeighley** [comment]
Looks good, you want me to bench test?

**Comment by henryhallam** [comment]
Sure if you have a chance to run it on a crappy computer, that'd be great.  But honestly I really don't expect this change to have any kind of noticeable performance impact.

**Comment by henryhallam** [comment]
Made quick quantitative CPU usage comparison, doesn't appear to be a difference although either way, the console is a CPU hog.

**Pull request merged**